### PR TITLE
fix(release_health): Adjust all code such that the current timestamp is determined exactly once [INGEST-784]

### PR DESCRIPTION
**Pull request by untitaker**
Basically pass around a bunch of `now` parameters where it makes sense.

Additionally I refactored tests such that the duplex backend is also tested. I verified that all tests run because pytest previously saw 68 of them, now it's 102, which is exactly 150%.

### CODE REVIEW

File: tests/snuba/sessions/test_sessions.py [hunk 1]
```diff
@@ -23,10 +52,7 @@ def make_24h_stats(ts):
     return _make_stats(datetime.utcfromtimestamp(ts).replace(tzinfo=pytz.utc), 3600, 24)
 
 
-class ReleaseHealthMetricsTestCase(SessionMetricsTestCase):
-    backend = MetricsReleaseHealthBackend()
-
-
+@parametrize_backend
 class SnubaSessionsTest(TestCase, SnubaTestCase):
     backend = SessionsReleaseHealthBackend()
```
Comment (jjbayer) [inline tests/snuba/sessions/test_sessions.py:57]: nit: Should we move this into `@parametrize_backend`?

Comment (untitaker) [reply tests/snuba/sessions/test_sessions.py:57]: yup ok!

File: src/sentry/release_health/duplex.py [hunk 2]
```diff
@@ -729,16 +731,25 @@ def should_compare(val: ReleaseSessionsTimeBounds) -> bool:
         )
 
     def check_has_health_data(
-        self, projects_list: Sequence[ProjectOrRelease]
+        self,
+        projects_list: Sequence[ProjectOrRelease],
+        now: Optional[datetime] = None,
```
Comment (jjbayer) [inline src/sentry/release_health/duplex.py:736]: Does the duplex impl actually need the `now` argument, or is it sufficient to add it to the sessions and metrics implementation? It's not part of the base class interface at `release_health.base`.

**Review by jjbayer — APPROVED** [review]

File: src/sentry/release_health/duplex.py (hunk 2 above)
Comment (untitaker) [reply src/sentry/release_health/duplex.py:736]: i think we shouldn't change the signature in subclasses, i just forgot to update the baseclass and apparently mypy can't be bothered to warn me about it either 🙃

**Pull request merged**